File: 백준/Silver/10828. 스택/스택.py
import sys
stack = []


def push(num):
    stack.append(num)


def pop():
    if (len(stack) == 0):
        print(-1)
    else:
        num = stack[-1]
        stack.pop(-1)
        print(num)


def size():
    print(len(stack))


def empty():
    if (len(stack) == 0):
        print(1)
    else:
        print(0)


def top():
    if (len(stack) == 0):
        print(-1)
    else:
        print(stack[-1])


N = int(input())

for i in range(N):
    ins = list(map(str, sys.stdin.readline().strip().split(' ')))
    a = str(ins[0])
    if a == "push":
        b = int(ins[1])
        push(b)
    elif a == "pop":
        pop()
    elif a == "size":
        size()
    elif a == "empty":
        empty()
    elif a == "top":
        top()

File: 백준/Silver/10828. 스택/스택.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = null;

        int N = Integer.parseInt(br.readLine());

        int[] arr = new int[N];

        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());

            String ins = st.nextToken();

            switch (ins) {
                case "push":
                    int num = Integer.parseInt(st.nextToken());
                    push(arr, num);
                    break;
                case "size":
                    System.out.println(size(arr));
                    break;
                case "empty":
                    System.out.println(empty(arr));
                    break;
                case "top":
                    System.out.println(top(arr));
                    break;
                case "pop":
                    System.out.println(pop(arr));
                    break;
            }
        }

    }

    public static void push(int[] arr, int num) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == 0) {
                arr[i] = num;
                break;
            }
        }
    }

    public static int size(int[] arr) {
        int cnt = 0;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] != 0) {
                cnt += 1;
            }
        }
        return cnt;
    }

    public static int empty(int[] arr) {
        if (size(arr) == 0) {
            return 1;
        } else {
            return 0;
        }
    }

    public static int pop(int[] arr) {
        if (empty(arr) == 1) {
            return -1;
        } else {
            int idx = size(arr) - 1;
            int num = arr[idx];
            arr[idx] = 0;
            return num;
        }
    }

    public static int top(int[] arr) {
        if (empty(arr) == 1) {
            return -1;
        } else {
            int idx = size(arr) - 1;
            return arr[idx];
        }
    }

}
